src/components/v2/Table/index.tsx

```diff
@@ -41,92 +23,88 @@ export const Table = ({
-        return orderDirection === 'asc' ? -1 : 1;
-      }
-      if (+a[rowIndex].value > +b[rowIndex].value) {
-        return orderDirection === 'asc' ? 1 : -1;
-      }
-      return 0;
-    });
-    return newRows;
-  }, [data, orderBy, orderDirection]);
-
-  /* gridStyles provides grid-template-columns styles for rows in rows for thead and tbody */
-  const gridStyles =
-    gridTemplateColumns ||
-    useMemo(
-      () =>
-        /* if gridTemplateColumns prop is not passed from parent component, we create similar fractions by default */
-        columns.map(() => '1fr').join(' '),
-      [columns],
-    );
+  const { rows, orderBy, orderDirection, onRequestOrder, gridColumns } = useTable({
+    columns,
+    initialOrder,
+    data,
+    gridTemplateColumns,
+    isMobileView,
+  });
 
   return (
-    <Paper css={styles.root} className={className}>
-      {title && <h4 css={styles.title}>{title}</h4>}
-
-      <TableContainer>
-        <TableMUI css={styles.table({ minWidth: minWidth ?? '0' })} aria-label={title}>
-          <Head
-            columns={columns}
-            orderBy={orderBy}
-            orderDirection={orderDirection}
-            onRequestOrder={onRequestOrder}
-            css={styles.getTemplateColumns({ gridStyles })}
-          />
-
-          {/* TODO: add loading state */}
-
-          {/* TODO: add error state */}
+    <Paper css={[styles.root, isMobileView && styles.rootMobile]} className={className}>
+      {title && <h4 css={[styles.title, isMobileView && styles.titleMobile]}>{title}</h4>}
 
-          <TableBody>
-            {rows.map(row => (
-              <TableRow
-                hover
-                key={row[rowKeyIndex].value.toString()}
-                onClick={e => rowOnClick && rowOnClick(e, row)}
-                css={styles.getTemplateColumns({ gridStyles })}
+      {isMobileView ? (
+        <>
+          {rows.map(([rowTitle, ...cells]) => (
+            <Paper key={uid(cells)} css={styles.tableWrapperMobile}>
+              <div css={styles.rowTitleMobile}>{rowTitle.render()}</div>
+              <Delimiter css={styles.delimiterMobile} />
+              <div
+                css={[
+                  styles.rowWrapperMobile,
+                  styles.getTemplateColumns({ gridColumns }),
+                  styles.getTemplateRows({ gridRows: gridTemplateRowsMobile }),
+                ]}
               >
-                {row.map(({ key, render }: ITableRowProps) => {
-                  const cellContent = render();
-                  const cellTitle = typeof cellContent === 'string' ? cellContent : undefined;
+                {cells.map(cell => {
+                  const currentColumn = columns.find(column => column.key === cell.key);
                   return (
-                    <TableCell css={styles.cellWrapper} key={uid(key)} title={cellTitle}>
-                      {cellContent}
-                    </TableCell>
+                    <div key={uid(cell)} css={styles.cellMobile}>
```

Comment:
As Corey mentioned, let's not use `uid`. Keys need to be based on an identifier that stays the same for a given cell between renders